flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -573,6 +576,23 @@ public static int getRunningContainers() {
 		return count;
 	}
 
+	void waitUntilApplicationFinished(ApplicationId applicationId, Duration timeout) throws Exception {
+		Deadline deadline = Deadline.now().plus(timeout);
+		while (true) {
```

Comment:
I would prefer
```
YarnApplicationState state;
do {
   state = yarnClient.getApplicationReport(applicationId).getYarnApplicationState();
   .....
} while (state != YarnApplicationState.FINISHED); 
```

Reply:
This may lead to unnecessary `sleep`, I change it to `while (state != YarnApplicationState.FINISHED) {}`

Reply:
agreed